### PR TITLE
Fix for scrollbar on touch devices.

### LINKED ISSUE #309: Touch / iOS scrolling not working as expected
## Expected Behavior

https://www.youtube.com/watch?v=9EVsS-YKPt8
For optimal user experience, touch scrolling should feel as close to native iOS 11 scrolling as possible.

## Current Behavior
Depending on settings(MOVE_AMPLITUDE, DECELERATION_AMPLITUDE...) scroll is either way too fast / "jumpy" or too slow.

Probably related to https://github.com/schrodinger/fixed-data-table-2/issues/220

Just changing above mentioned settings doesn't achieve expected behavior.

## Steps to Reproduce (for bugs)
Try to use touch scrolling with iOS 11. With default settings, even small "scroll event" moves down way too fast.

For example here:
http://schrodinger.github.io/fixed-data-table-2/example-touch-scroll.html

## Context
Can't really use fdt-2 in production, as big % of our visitors use latest iOS devices.

### PR DESCRIPTION
**Pull request by CharlesStover**
resolves #309

Clicking a vertical/horizontal scrollbar on a touch device triggers the table's touch event.
Moving the mouse/cursor/finger while touching a scrollbar causes it to drag in the opposite direction (intended behavior of touching the table), as opposed to causing the scrollbar to follow the mouse/cursor/finger.

## Description
This fixes that issue by preventing touch events from bubbling to the table-level touch handler if they are handled by child nodes (in this case, the scrollbars).

## Motivation and Context
Fixes unintuitive behavior of scrollbar moving in opposite direction of finger.

## How Has This Been Tested?
Chrome.

## Screenshots (if appropriate):
New behavior, as of this PR:

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -56,10 +56,11 @@
     "webpack-dev-server": "^2.6.1"
   },
   "scripts": {
+    "webpack": "webpack",
     "site-dev-server": "./build_helpers/startSiteDevServer.sh",
     "site-build": "./build_helpers/buildStaticSite.sh",
     "build-dist": "./build_helpers/buildDist.sh",
-    "build-npm": "./build_helpers/buildNPMInternals.sh",
```
[the remark this answers is not in the document]
Comment (CharlesStover) [reply package.json]: It did not work on Windows platforms. The shell script was just running Node, so turning it into a Node command made it platform-agnostic. As far as I'm aware, it should still work on Linux, while also working on Windows platforms now.

**Comment by CharlesStover** [comment]
>Would adding a touch handler to the scrollbar and having it stop propagation on the event work?

This should work. I definitely tried it and encountered an error, but I am starting to believe the error I encountered was the one where `position` is not using `getTouchX()`/`getTouchY()` and therefore didn't support touch events. I fixed that later, so the touch event listeners may work now. I'm testing it now.

**Comment by CharlesStover** [comment]
You may have some insight into this, but this is the behavior when using touch events (onTouchStart = preventDefault + onMouseMove):


[Here](https://github.com/schrodinger/fixed-data-table-2/compare/master...CharlesStover:touch-preventdefault?expand=1) is the branch for comparison.

**Comment by CharlesStover** [comment]
For the sake of discussion, this branch is otherwise exactly the same code as in the PR. The differences, to my knowledge, are that the PR immediately exits touch events on the table (early return), whereas this stops propagation of touch events on the scrollbar. Everything else is identical. This error did not occur in the original PR.
I would prefer this be implemented with stopPropagation as well, but I'm at a loss for the cause of this bug. Looks like maybe some sort of state issue, and I don't quite understand the intricacies of `_setNextState` or how else the scrollbar state may be managed. The code in the PR is in working order, however.

File: package.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (wcjordan) [reply package.json]: I'm happy to accept those changes, but could you do it as part of a separate PR and do it for all the build tools to keep things consistent?

**Comment by wcjordan** [comment]
I'm not sure I fully understand your previous comment / error you're describing, but in your touch-preventdefault branch it looks like you're only stopping propagation of the touchStart event.  You should also stop propagation for onTouchEnd, onTouchMove, and onTouchCancel if you want to keep the touch handler from coming into play.

Another alternative to look at might be to bind the touch handler in FixedDataTable.js to the rowsContainer (sibling of scrollbars) rather than to the top div (contains the scrollbars).

File: src/FixedDataTable.js [hunk 2]
```diff
@@ -608,6 +608,7 @@ var FixedDataTable = createReactClass({
   },
 
   _onRef(div) {
+    this._touchHandler.setRoot(div);
```
Comment (iwfafe) [inline src/FixedDataTable.js]: please check variable this._touchHandler is not null before call setRoot func. 
I have tested this and it throw error this._touchHandler is undefined.

**Comment by CharlesStover** [comment]
>You should also stop propagation for onTouchEnd, onTouchMove, and onTouchCancel if you want to keep the touch handler from coming into play.

Nailed it. I was under the impression that if touchStart never propagates to a parent element that touchMove would not trigger on that parent element.

File: src/Scrollbar.js [hunk 3]
```diff
@@ -195,7 +213,8 @@ var Scrollbar = createReactClass({
     this._mouseMoveTracker = new DOMMouseMoveTracker(
       this._onMouseMove,
       this._onMouseMoveEnd,
-      document.documentElement
+      document.documentElement,
+      true
```
Comment (wcjordan) [inline src/Scrollbar.js]: We should pass props.touchEnabled into Scrollbar and make this this.props.touchEnabled

**Comment by CharlesStover** [comment]
Should be all set.

**Comment by wcjordan** [comment]
Released w/ 0.8.14

**Pull request merged**